### PR TITLE
Refactoring/aws/cloudtrail

### PR DESCRIPTION
**Pull request by Aboisier**
This PR migrates CloudTrail to the new architecture, see #183 .

### CODE REVIEW

File: ScoutSuite/providers/aws/resources/cloudtrail/service.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+class CloudTrail(Regions):
+    _children = [
+        (Trails, 'trails')
+    ]
+
+    def __init__(self):
+        super(CloudTrail, self).__init__('cloudtrail')
+
+
+# # -*- coding: utf-8 -*-
```
Comment (Remi05) [inline ScoutSuite/providers/aws/resources/cloudtrail/service.py]: Is there a reason to keep the previous version commented?

Comment (Remi05) [reply ScoutSuite/providers/aws/resources/cloudtrail/service.py]: @Aboisier

Comment (Aboisier) [reply ScoutSuite/providers/aws/resources/cloudtrail/service.py]: The reason is I didn't push the commit 
which removed it lel

Comment (Remi05) [reply ScoutSuite/providers/aws/resources/cloudtrail/service.py]: lel

File: ScoutSuite/providers/aws/resources/cloudtrail/service.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+from ScoutSuite.providers.aws.resources.regions import Regions
+from ScoutSuite.providers.aws.resources.resources import AWSResources
+from ScoutSuite.providers.aws.facade.facade import AWSFacade
+from ScoutSuite.providers.base.utils import get_non_provider_id
```
Comment (misg) [inline ScoutSuite/providers/aws/resources/cloudtrail/service.py]:
```suggestion
from ScoutSuite.providers.utils import get_non_provider_id
```

Comment (misg) [reply ScoutSuite/providers/aws/resources/cloudtrail/service.py]: (There's already a `utils.py` file defining `get_non_provider_id` there.)

File: ScoutSuite/providers/base/utils.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from hashlib import sha1
```
Comment (misg) [inline ScoutSuite/providers/base/utils.py]: You can get rid of this file.

File: ScoutSuite/providers/utils.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
-        """
-        _hash = sha1()
-        _hash.update(name.encode('utf-8'))
-        return _hash.hexdigest()
+    :param name:                    Name of the resource to
+    :return:                        SHA1(name)
+    """
+    hash = sha1()
+    hash.update(name.encode('utf-8'))
+    return hash.hexdigest()
```
Comment (Aboisier) [inline ScoutSuite/providers/utils.py]: No hate, the indentation was broken, and I think using _ or local variables is not PEP 8 compliant.

Comment (misg) [reply ScoutSuite/providers/utils.py]: `hash` is a python built-in.

Comment (Aboisier) [reply ScoutSuite/providers/utils.py]: lol

**Pull request merged**